File: AcePump.WebApi/VirtualPathMapper.cs
using System;
using System.Web.Hosting;
using AcePump.Common;

namespace AcePump.WebApi
{
    public class VirtualPathMapper : IVirtualPathMapper
    {

        private static Lazy<VirtualPathMapper> Ctor = new Lazy<VirtualPathMapper>(()  => new VirtualPathMapper());

        public static VirtualPathMapper Instance
        {
            get
            {
                return Ctor.Value;
            }
        }

        private VirtualPathMapper()
        {
        }

        public string MapPath(string appRelativePath)
        {
            return HostingEnvironment.MapPath(appRelativePath);
        }
    }
}

File: AcePump.Web/VirtualPathMapper.vb
Imports AcePump.Common

Public Class VirtualPathMapper
    Implements IVirtualPathMapper

    Private Shared Ctor As Lazy(Of VirtualPathMapper) = New Lazy(Of VirtualPathMapper)(Function() New VirtualPathMapper())
    Public Shared ReadOnly Property Instance As VirtualPathMapper
        Get
            Return Ctor.Value
        End Get
    End Property

    Private Sub New()
    End Sub

    Public Function MapPath(appRelativePath As String) As String Implements IVirtualPathMapper.MapPath
        Return Hosting.HostingEnvironment.MapPath(appRelativePath)
    End Function
End Class
